### ISSUE TITLE
ELM crashes with NaN col_pf%col_plant_pdemand_vr

### ISSUE THREAD

**Issue by tanzeli1982** [E3SM-Project/E3SM#2996]
@qzhu-lbl @jinyuntang @bishtgautam 

ELM crashes due to **NaN** col_pf%col_plant_pdemand_vr() values when using the compset of I20TRGSWCNPECACNTBC or I20TRCNPECACNTBC. See the dumped stack and my variable outputs below:

  1:  index                      =         5267           1
  1:  gross_pmin_vr              =   2.963092358542597E-009
  1:  potential_immob_p_vr       =   1.200258621398698E-012
  1:  col_plant_pdemand_vr       =                      NaN
  1:  biochem_pmin_vr_col        =   0.000000000000000E+000
  1:  primp_to_labilep_vr_col    =   0.000000000000000E+000
  1:  pdep_to_sminp              =   2.497044401570777E-011
  1:  ndep_prof                  =    57.1010333563630     
 60: forrtl: error (65): floating invalid
 60: Image              PC                Routine            Line        Source             
 60: e3sm.exe           00000000064A041E  Unknown               Unknown  Unknown
 60: e3sm.exe           0000000005D144F0  Unknown               Unknown  Unknown
 60: e3sm.exe           00000000010E5007  allocationmod_mp_        2423  AllocationMod.F90
 60: e3sm.exe           0000000003153BBD  soillittdecompmod         410  SoilLittDecompMod.F90
 60: e3sm.exe           00000000022C927E  ecosystemdynmod_m         589  EcosystemDynMod.F90
 60: e3sm.exe           000000000090D181  clm_driver_mp_clm        1028  clm_driver.F90
 60: e3sm.exe           0000000000873E46  lnd_comp_mct_mp_l         509  lnd_comp_mct.F90
 60: e3sm.exe           000000000046667F  component_mod_mp_         737  component_mod.F90
 60: e3sm.exe           000000000043011F  cime_comp_mod_mp_        2603  cime_comp_mod.F90
 60: e3sm.exe           000000000044E7E4  MAIN__                    133  cime_driver.F90
 60: e3sm.exe           000000000040A80E  Unknown               Unknown  Unknown
 60: e3sm.exe           0000000006574539  Unknown               Unknown  Unknown
 60: e3sm.exe           000000000040A6F9  Unknown               Unknown  Unknown

**Comment by jinyuntang** [comment]
Did you rebase with the master? I remember I once fixed a similar error.

**Comment by bishtgautam** [comment]
Or, does this BGC compset currently works only on maint-1.1?

**Comment by tanzeli1982** [comment]
My head tag is 59ce0c85a. Did you commit something newer?

**Comment by tanzeli1982** [comment]
Gautam, which BGC compset do you know works widely? I can have a try.

**Comment by bishtgautam** [comment]
One second thought, I20TRGSWCNPECACNTBC should work because we have a test for it using next and master. see https://my.cdash.org/viewTest.php?onlypassed&buildid=1664003

**Comment by tanzeli1982** [comment]
@bishtgautam @jinyuntang @qzhu-lbl 
I have checked why other listed variables (**gross_pmin_vr**, **potential_immob_p_vr**, **biochem_pmin_vr_col**, **primp_to_labilep_vr_col**, **pdep_to_sminp**) did not have NaN values even though they are set in the same way in **col_pf_setvalues()**. The reason is obviously that they are set before using in other places (e.g., ./biogeochem/PhosphorusDynamicsMod.F90).

**Comment by tanzeli1982** [comment]
@bishtgautam I closed it because the error is caused by surface data. I am not pretty sure of the reason but mksurfdata_map cannot reproduce the default dataset (surfdata_360x720cru_simyr2000_c180216.nc) even by using the default raw data and mapping files. So currently I just tried the following workaround: I copy all data in surfdata_360x720cru_simyr2000_c180216.nc to a new nc file and then insert my own data into it.
